### PR TITLE
use round-cornered account image

### PR DESCRIPTION
**Pull request by Chilledheart**

### CODE REVIEW

**Comment by Chilledheart** [comment]
(from Windows 7)

screen shot 2015-02-09 at 15 28 17

**Comment by Chilledheart** [comment]
Updated version
screen shot 2015-02-10 at 15 22 00
screen shot 2015-02-10 at 15 23 15

**Comment by Chilledheart** [comment]
screen shot 2015-02-10 at 16 08 10
Improved with retina support, inspired by https://qt.gitorious.org/qt-creator/qt-creator/commit/3779147ffc1b5876565dfb85c0219e65111ee0c9

File: src/ui/events-list-view.cpp [hunk 1]
```diff
@@ -100,16 +101,32 @@ void EventItemDelegate::paint(QPainter *painter,
...
+    mask_painter.setRenderHint(QPainter::Antialiasing);
+    mask_painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
+    mask_painter.setBrush(Qt::white);
+    mask_painter.drawEllipse(actualRect);
+    mask_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
+    mask_painter.drawImage(actualRect, avatar);
+    mask_painter.setCompositionMode(QPainter::CompositionMode_DestinationOver);
+    mask_painter.fillRect(actualRect, Qt::transparent);
+    mask_painter.end();
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
```
Comment (lins05) [inline src/ui/events-list-view.cpp]: The above code manipulating the mask image is the same as those in `accounts-view.cpp`, can we refactor them out to a common function?

**Comment by lins05** [comment]
LGTM, just some of the common code can be refactored out as I wrote in the line comment.

File: src/ui/events-list-view.cpp (hunk 1 above)
Comment (Chilledheart) [inline src/ui/events-list-view.cpp]: Please don't. There are only one place and the code looks similar but serves on different purposes. I would rather to remain what they are.

Comment (lins05) [inline src/ui/events-list-view.cpp]: The code between line 109 to 121 is the same as those in line 214 to 226 of account-view.cpp except for one argument (`image` v.s. `avatar`). It's better to refactor them out to make it more maintainbale, and reusable latter.

```cpp
// events-list-view.cpp
QImage masked_image(actualRect.size(), QImage::Format_ARGB32_Premultiplied);
masked_image.fill(Qt::transparent);
QPainter mask_painter;
mask_painter.begin(&masked_image);
mask_painter.setRenderHint(QPainter::Antialiasing);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
mask_painter.setBrush(Qt::white);
mask_painter.drawEllipse(actualRect);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
mask_painter.drawImage(actualRect, avatar);
mask_painter.setCompositionMode(QPainter::CompositionMode_DestinationOver);
mask_painter.fillRect(actualRect, Qt::transparent);
mask_painter.end();
```
```cpp
// account-view.cpp
QImage masked_image(actualRect.size(), QImage::Format_ARGB32_Premultiplied);
masked_image.fill(Qt::transparent);
QPainter mask_painter;
mask_painter.begin(&masked_image);
mask_painter.setRenderHint(QPainter::Antialiasing);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
mask_painter.setBrush(Qt::white);
mask_painter.drawEllipse(actualRect);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
mask_painter.drawPixmap(actualRect, image);
mask_painter.setCompositionMode(QPainter::CompositionMode_DestinationOver);
mask_painter.fillRect(actualRect, Qt::transparent);
mask_painter.end();
```

**Comment by lins05** [comment]
@freeplant Is this ui ready for merging?

**Comment by freeplant** [comment]
No, I haven't compile it on my computer and haven't seen the full screenshot.

**Comment by freeplant** [comment]
@Chilledheart Can you upload a screenshot of the whole main-panel.

**Comment by Chilledheart** [comment]
@freeplant Here is a screenshot of main windows.Is that what you want?
screenshot from 2015-02-15 16 35 51

**Pull request merged**